comm/src/main/scala/coop/rchain/comm/transport/GrpcTransportClient.scala

```diff
@@ -46,12 +47,12 @@ class GrpcTransportClient(
   private def certInputStream  = new ByteArrayInputStream(cert.getBytes())
   private def keyInputStream   = new ByteArrayInputStream(key.getBytes())
   private val streamObservable = new StreamObservable(clientQueueSize, tempFolder)
-  private val parallelism      = Math.max(Runtime.getRuntime.availableProcessors(), 2)
   private val streamQueue =
     streamObservable
       .flatMap { s =>
         Observable
           .fromIterable(s.peers)
+          .executeOn(ioScheduler)
```

Comment:
This is not simply io operation. That why previously this was fixed pool with the number of proc size. This should be new on the main scheduler. Parallelism as the number of proc is fine because Observable does not create real OS threads but just Task instances that can be run concurrently if there is not enough parallel threads.